macrosynergy/pnl/contract_signals.py

```diff
@@ -69,17 +65,20 @@ def _check_arg_types(
         ]
     )
 
-    non_empty_iterables: bool = all(
-        [
-            df is None or len(df) > 0,
-            cids is None or len(cids) > 0,
-            ctypes is None or len(ctypes) > 0,
-            cscales is None or len(cscales) > 0,
-            csigns is None or len(csigns) > 0,
-            hbasket is None or len(hbasket) > 0,
-            hscales is None or len(hscales) > 0,
-        ]
-    )
+    try:
```

Comment:
If I am honest, I think we are trying to be too clever for the data-type checking.

Personally I would prefer something like: `assert df is None or (isinstance(df, pd.DataFrame) and not df.empty)`. Although it takes up more lines, being more explicit about what we are checking for it useful.